cli/flox-rust-sdk/src/models/pkgdb.rs

```diff
@@ -89,7 +91,21 @@ pub enum CallPkgDbError {
 ///
 /// Error JSON is parsed into a [CallPkgDbError::PkgDbError].
 pub fn call_pkgdb(mut pkgdb_cmd: Command) -> Result<Value, CallPkgDbError> {
+    // Configure pkgdb PATH with exact versions of everything it needs.
+    //
+    // Nix itself isn't pure, which is to say that it isn't built with a
+    // reference to `git` in its closure, so correspondingly depends upon
+    // finding it in $PATH to function.
+    //
+    // This just isn't OK for us as we're looking for flox to operate reliably
+    // in "hostile" environments, which includes situation where a user may
+    // redefine or blat their $PATH variable entirely, so we always invoke
+    // pkgdb with an explicit PATH of our making.
+    //
+    // It really shouldn't be necessary to append $PATH, so we won't.
+    let pkgdb_path = Path::new(&*GIT_PKG_BIN);
```

Comment:
nonblocking: this should be in a separate PR. Especially since we squash and merge

Reply:
Talked with Matthew ... ideally would have merged this before 1299 but this was a weird one in that the refactor was the thing that surfaced the issue. Resolving convo.